UI/src/view/point_asset/trafficLayer.js

```diff
@@ -0,0 +1,53 @@
+(function(root) {
+    root.TrafficLayer = function(params) {
+        PointAssetLayer.call(this, params);
+
+        var me = this;
+        var application= applicationModel,
+            map = params.map,
+            mapOverlay = params.mapOverlay,
+            roadCollection = params.roadCollection,
+            selectedAsset = params.selectedAsset;
+
+         this.handleMapClick = function (coordinates) {
+            if (application.getSelectedTool() === 'Add' && zoomlevels.isInAssetZoomLevel(zoomlevels.getViewZoom(map))) {
+                me.createNewAsset(coordinates);
+            } else if (selectedAsset.isDirty()) {
+                me.displayConfirmMessage();
+            }
+        };
+
+        this.createNewAsset = function(coordinates) {
```

Comment:
The only difference with this function and the parent one is a different road collection parameter for excluding road links. It could probably be shortened in some way.